File: 2323-minimum-bit-flips-to-convert-number/minimum-bit-flips-to-convert-number.java
class Solution {
    public int minBitFlips(int start, int goal) {
        if (start == goal)
            return 0;
        
        String sbin = Integer.toBinaryString(start);
        String ebin = Integer.toBinaryString(goal);

        int s = sbin.length(), e = ebin.length();
        int difference = Math.abs(s - e);
        if (s < e) {
            for (int i = 0; i< difference; i++) 
                sbin = "0" + sbin;
        } else  {
            for (int i = 0; i< difference; i++)
                ebin = "0" + ebin;
        }

        int count = 0;
        for (int i = 0; i < sbin.length(); i++)
            if (sbin.charAt(i) != ebin.charAt(i))
                count += 1;
        return count;
        
    }
}

File: 2323-minimum-bit-flips-to-convert-number/minimum-bit-flips-to-convert-number.py
class Solution:
    def minBitFlips(self, start: int, goal: int) -> int:
        if start == goal:
            return 0

        sbin = str(bin(start))[2:]
        ebin = str(bin(goal))[2:]

        s, e = len(sbin), len(ebin)
        difference = abs(s - e)
        if s < e:
            sbin = '0' * difference + sbin
        else:
            ebin = '0' * difference + ebin 

        count = 0
        for i in range(len(sbin)):
            if sbin[i] != ebin[i]:
                count += 1
        return count